### ISSUE TITLE
Detouring a function with a reference parameter

### ISSUE THREAD

**Issue by whatisaphone** [darfink/detour-rs#9]
Something goes wrong deep in the innards of the macro when you try to detour an extern "Rust" function that takes a reference as a parameter:

    static_detours! {
        struct DetourPrintln: fn(&str);
    }

```
error[E0308]: mismatched types
  --> crates\core\src\lib.rs:21:5
   |
21 | /     static_detours! {
22 | |         pub struct DetourPrintln: fn(&str);
23 | |     }
   | |_____^ one type is more general than the other
   |
   = note: expected type `detour::Function`
              found type `detour::Function`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

**Comment by darfink** [comment]
That's correct. The `Function` trait cannot express lifetime semantics of individual arguments and has therefore a `'static` requirement.

An alternative is to use the more cumbersome `RawDetour`.

Perhaps there is a potentially better alternative implementation but I've yet to identify a solution that is type-safe.

**Comment by whatisaphone** [comment]
I understand. That's an unfortunate limitation. Thanks for the explanation and workaround 👍

**Comment by darfink** [comment]
Still a limitation of the typed detours in [0.9.0](https://github.com/darfink/detour-rs/releases/tag/v0.9.0). `RawDetour` works for such signatures (now tested and shown in the README); a macro-generated signature type could lift this in the future.
